### ISSUE TITLE
Proposal: Adopt unitsSI from openPMD

### ISSUE THREAD

**Issue by benajamin** [nexusformat/definitions#1325]
[openPMD](https://www.openpmd.org/#/start) [describes a unit](https://github.com/openPMD/openPMD-standard/blob/latest/STANDARD.md#unit-systems-and-dimensionality) by a list of exponents describing the magnitude of the possible dimensions: length L, mass M, time T, electric current I, thermodynamic temperature theta, amount of substance N, luminous intensity J. There was a request from the recent [HELPMI](https://helmholtz-metadaten.de/de/inf-projects/helpmi-helmholtz-laser-plasma-metadata-initiative) [workshop](https://indico.gsi.de/event/17826/overview) that the HELPMI community be able to describe units in the openPMD format.

[Currently, NeXus expects](https://manual.nexusformat.org/datarules.html#design-units) an `NX_CHAR` string in the `@units` attribute, which is apparently not being validated. Allowing the openPMD way of describing units could be as simple as allowing the content of `@units` to be either an `NX_CHAR` string (for the current NeXus way), or an array of 7 signed floats (for the openPMD style). I expect that open PMD chose to allow floats instead of just integers so that strange units could be accommodated (e.g. imperial measures, or relativistic units based on _c_). Validation of the openPMD style units would be fairly simple since you could just check that the array values have the correct pattern of zero or non-zero values.

Do others in the community like this idea? Is there a better way to implement it in NeXus?

**Comment by benajamin** [comment]
One issue with this system is that it doesn't specify the power of the dimension, i.e. meters of length and cubic meters of volume are not differentiated. Being able to specify the dimension power would help to differentiate similar measures, such as the flux measurements described in NXmx. Specifying this would require an extra set of numbers.

**Comment by benajamin** [comment]
I agree with @phyy-nx  that this would add computer readability at the expense of human readability. This is perhaps mitigated by viewers being able to interpret the openPMD dimensions vector into a corresponding human understandable unit string.

**Comment by benajamin** [comment]
This idea is referenced in discussion of #1435

**Comment by prjemian** [comment]
I believe, strongly, that this type of representation has its best use internal to a program. It should not appear in interfaces with humans.

**Comment by prjemian** [comment]
Our NXDL files are human-facing in several ways.

- contents appear in published documentation
- our community is expected to compose, edit, and review NXDL files

The vector representation introduces complexity where a simpler solution is
available.
